examples/gamepad.html

```diff
@@ -92,68 +124,95 @@
           audioElement.play();
       };
 
-      var rotator;
-      
       // Initialize the decoder and activate the button when ready.
       decoder.initialize().then(function () {
         buttonPlayback.disabled = false;
         selectSource.disabled = false;
-        rotator = Omnitone.createFOARotator(audioContext);
         window.requestAnimationFrame(runAnimation);
       }, function (onInitializationError) {
         console.error(onInitializationError);
       });
 
-      var textRotation = document.getElementById('rotation');
+      var textRotation = document.getElementById('eRotation');
+      var divRotator = document.getElementById('eRotator');
 
+      // called if HTML selector is changed
+      selectRotation = function (value) {
+        var direction = predefinedDirections[value];
+        updateDirection(direction[0], direction[1], direction[2]);
+      };
+
+      // sets the decoder's rotation matrix and displays the angles
+      // Tait–Bryan angles represent rotations about three distinct axes
+      // z-y’-x″ (intrinsic rotations) are known as: yaw, pitch and roll
+      function updateDirection(yaw, pitch, roll) {
+
+        // http://mathworld.wolfram.com/EulerAngles.html
+        // theta is pitch, psi is roll, and  phi is yaw.
+        // but do not ask me why this decoder uses a different convention:
+        var theta = yaw;
+        var psi = pitch;
+        var phi = roll;
+
+        decoder.setRotationMatrix([
+          Math.cos(theta) * Math.cos(phi),
+          Math.cos(theta) * Math.sin(phi),
+          -Math.sin(theta),
+          Math.sin(psi) * Math.sin(theta) * Math.cos(phi)
+            - Math.cos(psi) * Math.sin(phi),
+          Math.sin(psi) * Math.sin(theta) * Math.sin(phi)
+            + Math.cos(psi) * Math.cos(phi),
+          Math.cos(theta) * Math.sin(psi),
+          Math.cos(psi) * Math.sin(theta) * Math.cos(phi)
+            + Math.sin(psi) * Math.sin(phi),
+          Math.cos(psi) * Math.sin(theta) * Math.sin(phi)
+            - Math.sin(psi) * Math.cos(phi),
+          Math.cos(theta) * Math.cos(psi)
+        ]);
+
+        textRotation.innerHTML = 'yaw ' + Math.round(yaw * 180 / Math.PI)
+          + ' pitch ' + Math.round(pitch * 180 / Math.PI)
+          + ' roll ' + Math.round(roll * 180 / Math.PI);
+      }
+
+      selectRotation(0);
+
+      var gamepad = false;
+      var axes = [ 0,0,0 ];
+
+      // check checks for gamepads or query the existing gamepad for changes
       function runAnimation() {
 
         // look for a Gamepad and check, whether is has changed it axes values.
-        var gamepads = navigator.getGamepads ? navigator.getGamepads() : (navigator.webkitGetGamepads ? navigator.webkitGetGamepads() : []);
-        var changed = false;
+        var gamepads = navigator.getGamepads ? navigator.getGamepads() :
+          (navigator.webkitGetGamepads ? navigator.webkitGetGamepads()
+            : []);
         for (var i = 0; i < gamepads.length; ++i) {
```

Comment:
This looks like you're handling multiple gamepad in a single update. So if there are multiple pads connected and actively moving, what's going to happen?